### PR TITLE
feat: update cachecontrol to 0.13.0

### PR DESCRIPTION
**Pull request by ralbertazzi**
# Pull Request Check List

Closes: https://github.com/python-poetry/poetry/pull/7997

@Secrus didn't want to step on your toes, just provide help :)

### CODE REVIEW

**Comment by popadi** [comment]
Initially I hoped that a new cachecontrol release will silently fix the issue described [here](https://github.com/python-poetry/poetry/issues/7936) (without having to update to a newer version of poetry OR to pin urllib3). However, because the toml file has `^0.12.9` for all releases and because the new cachecontrol was release as `0.13.0`, this won't happen. 

What will be the poetry versions that will be lucky to benefit from this?

**Comment by ralbertazzi** [comment]
> without having to update to a newer version of poetry OR to pin urllib3

I don't think anyone will release a fix for older versions of Poetry due to the limited time that maintainers can invest. The issue has been closed since fixed in latest versions, please update to 1.5

**Comment by dimbleby** [comment]
recent versions of poetry all pin to urrlib3<2, therefore if correctly installed they are not exposed to that issue anyway.

however this reminds me that this MR should allow that constraint to be relaxed, urllib3 version 3 is now out.

**Comment by ralbertazzi** [comment]
Could you point me to the release? I can only see 2.0.2

**Comment by dimbleby** [comment]
yes sorry, I meant 2.  but either way, this MR should allow that to be relaxed

File: pyproject.toml [hunk 1]
```diff
@@ -60,7 +58,6 @@ tomlkit = ">=0.11.4,<1.0.0"
 trove-classifiers = ">=2022.5.19"
 virtualenv = "^20.22.0"
 xattr = { version = "^0.10.0", markers = "sys_platform == 'darwin'" }
-urllib3 = "^1.26.0"
```
Comment (dimbleby) [inline pyproject.toml:63]: poetry still has a direct dependency on `urllib3` - https://github.com/python-poetry/poetry/blob/50e61e696bb0ae61c148d570f1c204a667464f1c/src/poetry/publishing/uploader.py#L20 - so this should not be removed.

Comment (ralbertazzi) [reply pyproject.toml:63]: Seemed really minor for me to be kept but I understand the reasoning, better to be safe than sorry!

Comment (ralbertazzi) [reply pyproject.toml:63]: Would you prefer to relax the constraint or pin it to `^2.0.2` to enforce the usage of a more recent version? I'd prefer the second but I also understand the Poetry direct dependency on urllib3 works ok with <2 too..

Comment (dimbleby) [reply pyproject.toml:63]: so far as I can see that import is only used to set up a `Retry`, which is configured to retry only for GETs; and the uploader never does a GET, it only POSTs.

So if that's right then that little pile of code could be deleted anyway.   But double-check, I only skimmed it..!

Comment (ralbertazzi) [reply pyproject.toml:63]: I think you are totally right. Wouldn't it be better to then fix it to "POST" if the original code was to retry uploads?

Comment (dimbleby) [reply pyproject.toml:63]: meh, if the code has been there for this long without anyone feeling the need to have working retries - I'd just delete it.

Comment (ralbertazzi) [reply pyproject.toml:63]: I was just writing that 😄 Makes sense

**Review by Secrus — APPROVED** [review]

**Comment by frostming** [comment]
> However, because the toml file has `^0.12.9` for all releases and because the new cachecontrol was release as `0.13.0`, this won't happen.

I've released v0.12.13 with the necessary fixes

**Pull request merged**